clients/client-java/src/main/java/com/datastrato/gravitino/client/HTTPClient.java

```diff
@@ -314,6 +319,23 @@ private <T> T execute(
       Map<String, String> headers,
       Consumer<ErrorResponse> errorHandler,
       Consumer<Map<String, String>> responseHeaders) {
+
+    if (beforeConnectHandle != null) {
+      synchronized (this) {
+        if (beforeConnectHandle != null) {
+          Function<HTTPClient, Void> handle = beforeConnectHandle;
+          beforeConnectHandle = null;
```

Comment:
What is the purpose of using double lock to check the variable?

Reply:
Double-checking is used to avoid unnecessary locking overhead after a successful connection